appservice/src/TunnelProxy.ts

```diff
@@ -141,20 +141,20 @@ class RetryableTunnelStatusError extends Error { }
  * A local TCP server that forwards all connections to the Kudu tunnel websocket endpoint.
  */
 export class TunnelProxy {
-    private _localPort: number;
-    private _remotePort: number;
+    private _port: number;
     private _client: SiteClient;
     private _publishCredential: User;
     private _server: Server;
     private _openSockets: TunnelSocket[];
+    private _useDefaultPort: boolean;
 
-    constructor(localPort: number, remotePort: number, client: SiteClient, publishCredential: User) {
-        this._localPort = localPort;
-        this._remotePort = remotePort;
+    constructor(port: number, client: SiteClient, publishCredential: User, useDefaultPort: boolean = false) {
```

Comment:
I think `sshPort` is more accurate than `defaultPort`. It's also not really "using" that port - it's just validating that port. I'd prefer a variable name like `isSsh`.